fre/cmor/cmor_mixer.py

```diff
@@ -1,408 +1,596 @@
-#!/usr/bin/env python
 '''
-see README.md for CMORmixer.py usage
+python module housing the metadata processing routines utilizing the cmor module, in addition to
+click API entry points
+see README.md for additional information on `fre cmor run` (cmor_mixer.py) usage
 '''
 
-# TODO : reconcile 'lst' variable names with 'list' in variable names
-#        as this is confusing to read and ambiguous to interpret
-#        probably good to avoid the word 'list' in the names
-# variable ierr is unused... what is it and what does it do?
-#        commented out until further investigation done
-
 import os
 import json
+import subprocess
+from pathlib import Path
 
 import netCDF4 as nc
 import click
 import cmor
 
 
+# ----- \start consts
+
+# ----- \end consts
+
+### ------ helper functions  ------ ###
 def copy_nc(in_nc, out_nc):
-    ''' copy a net-cdf file, presumably '''
-    print('\n\n----- START copy_nc call -----')
+    '''
+    copy target input netcdf file in_nc to target out_nc. I have to think this is not a trivial copy
+    operation, as if it were, using shutil's copy would be sufficient. accepts two arguments
+        in_nc: string, path to an input netcdf file we wish to copy
+        out_nc: string, an output path to copy the targeted input netcdf file to
+    '''
     print(f'(copy_nc)  in_nc: {in_nc}')
     print(f'(copy_nc)  out_nc: {out_nc}')
+
     # input file
     dsin = nc.Dataset(in_nc)
 
-    # output file
-    dsout = nc.Dataset(out_nc,
-                       "w") #, format = "NETCDF3_CLASSIC")
+    # output file, same exact data_model as input file.
+    # note- totally infuriating...
+    #       the correct value for the format arg is netCDF4.Dataset.data_model
+    #       and NOT netCDF4.Dataset.disk_format
+    dsout = nc.Dataset(out_nc, "w",
+                       format = dsin.data_model)
```

Comment:
i wanted to respect the fileformat as a default choice for now! hard to understand the original scripts full intentions with the commented-out netcdf3 classic specification...

is output format specification desired? such that a specific output target format can be written out? I think the `cmor` module may enforce netcdf4 anywho...

Reply:
Respecting the input format is usually the right thing to do, but for CMOR it might be worth only supporting Netcdf-4 with mild compression (as we did for CMIP6).